### PR TITLE
2032 Native classes generating

### PR DESCRIPTION
**Pull request by levBagryansky**
Closes: #2032 
Create java classes with native method insyde for every rust insert.
This class then is placed to `generated-sources/EOrust/natives` package.

---

## PR-Codex overview
This PR refactors the Rust code generation and improves the build process. It also updates some dependencies and adds a new method to the `EOrust` class. 

### Detailed summary
- Refactors Rust code generation and improves build process
- Updates dependencies
- Adds new method to `EOrust` class for deserializing objects from base64-encoded strings

> The following files were skipped due to too many changes: `eo-runtime/src/main/java/EOorg/EOeolang/EOrust.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by levBagryansky** [comment]
@Graur please take a look

**Review by Graur — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/test/java/org/eolang/maven/rust_project/NativeTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Test case for {@link Native}.
+ *
+ * @since 0.1
+ */
+final class NativeTest {
+
+    @Test
+    void savesCorrectly(@TempDir final Path temp) throws Exception {
+        final Path target = Paths.get("mypackage");
+        final Native klass = new Native("name", target.toString());
```
[the remark this answers is not in the document]
Comment (levBagryansky) [reply eo-maven-plugin/src/test/java/org/eolang/maven/rust_project/NativeTest.java]: Indeed

**Comment by levBagryansky** [comment]
@Graur please take a look

**Comment by levBagryansky** [comment]
@Graur please check

File: eo-maven-plugin/src/test/java/org/eolang/maven/rust/ModuleTest.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ void transformsCorrectly(@TempDir final Path temp) throws Exception {
                 "use jni::sys::{jint};",
                 "use jni::JNIEnv;",
                 "#[no_mangle]",
-                "pub extern \"system\" fn Java_EOorg_EOeolang_EOrust_simple(_env: JNIEnv, _class: JClass,) -> jint {"
+                "pub extern \"system\" fn Java_EOrust_natives_simple_simple(_env: JNIEnv, _class: JClass,) -> jint {"
```
Comment (Graur) [inline eo-maven-plugin/src/test/java/org/eolang/maven/rust/ModuleTest.java:65]: @levBagryansky Just curious why the name was formed like that?

Comment (levBagryansky) [reply eo-maven-plugin/src/test/java/org/eolang/maven/rust/ModuleTest.java:65]: @Graur This is how the function must be named depending on the java functon name. For example, if the class `Cat` has native method `say` in package `com.org`, the name will be smth like `Java_com_org_Cat_say`

**Review by Graur — APPROVED** [review]
@yegor256 Please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**